src/wallet/rpcwallet.cpp

```diff
@@ -2876,6 +2877,27 @@ static UniValue listunspent(const JSONRPCRequest& request)
                 CScript redeemScript;
                 if (pwallet->GetCScript(hash, redeemScript)) {
                     entry.pushKV("redeemScript", HexStr(redeemScript.begin(), redeemScript.end()));
+                    // Now check if the redeemScript is actually a P2WSH script
+                    CTxDestination witness_destination;
+                    if (redeemScript.IsPayToWitnessScriptHash()) {
+                        assert(ExtractDestination(redeemScript, witness_destination));
```

Comment:
In commit "Make listunspent return witness script for P2WSH or P2SH-P2WSH" (0b665ab577ab41e3b7b12d9b7d83c151cf90fdaf)

This would be broken if compiled with NDEBUG, which even though we don't support it, isn't really great for readability because you don't expect to see code with side effects inside an assert. Would be a little better to write something like:

```c++
bool extracted = ExtractDestitionation...;
assert(extracted);
```